src/Players/AIPlayer.hs

```diff
@@ -79,10 +79,28 @@ minimize (GameTree chessBoard score children) colour = (GameTree chessBoard scor
 
 
 -- Returns the next move that maximizes the score
--- TODO: implement + test this [Yiyi]
-getMoveWithMaxScore :: GameTree -> ChessMove
-getMoveWithMaxScore gameTree = ChessMove ('z', -1) ('z', -1)
-
+-- TODO: test this [Yiyi]
+getMoveWithMaxScore :: GameTree -> ChessPieceColour -> ChessMove
+getMoveWithMaxScore (GameTree _ _ nextMoves) colour = moveOfMaxScore
+    where movesWithScore = zip allMoves scoreOfMoves
+          allMoves = [move | (MoveSubtree move _) <- nextMoves]              -- nextMoves :: [MoveSubtree]
+          scoreOfMoves = map (\ tree -> getScore tree) subGameTrees
+          subGameTrees = [gameTree | (MoveSubtree _ gameTree) <- nextMoves]
+          moveWithMaxScore = foldr (\ acc scoreOfMove -> if (isGreater scoreOfMove acc) then scoreOfMove else acc) (head movesWithScore) (tail movesWithScore)
```

Comment:
I think `foldr` passes the accumulator as the second argument - maybe double check?

Reply:
Recall that the move with the max score is also the move with the same score as the root (because we've called `maximize` on it). So you can just check which move has the same score as the root:

```haskell
moveWithMaxScore = (filter (\ (move, score) -> score == rootScore) movesWithScore) !! 0
```

Reply:
You are right! I double checked, it's `foldl` passes the acc as the 1st argument. But I changed it to use the `filter` above.